@navikt/core/react/src/overlays/floating-menu/Menu.tsx

```diff
@@ -0,0 +1,1183 @@
+    | "side"
+    | "align"
+  > {}
+
+const MenuSubContent = forwardRef<
+  MenuContentInternalElement,
+  MenuSubContentProps
+>((props: MenuSubContentProps, forwardedRef) => {
+  const descendants = useMenuDescendants();
+
+  const context = useMenuContext();
+  const rootContext = useMenuRootContext();
+  const subContext = useMenuSubContext();
+  const ref = useRef<MenuContentInternalElement>(null);
+  const composedRefs = useMergeRefs(forwardedRef, ref);
+
+  return (
+    <MenuDescendantsProvider value={descendants}>
+      <MenuContentInternal
+        id={subContext.contentId}
+        aria-labelledby={subContext.triggerId}
+        {...props}
+        ref={composedRefs}
+        align="start"
+        side="right"
+        disableOutsidePointerEvents={false}
+        onOpenAutoFocus={(event) => {
+          // when opening a submenu, focus content for keyboard users only
+          if (rootContext.isUsingKeyboardRef.current) {
+            ref.current?.focus();
+          }
+          event.preventDefault();
+        }}
+        // The menu might close because of focusing another menu item in the parent menu. We
+        // don't want it to refocus the trigger in that case so we handle trigger focus ourselves.
+        onCloseAutoFocus={(event) => event.preventDefault()}
+        onFocusOutside={composeEventHandlers(props.onFocusOutside, (event) => {
+          // We prevent closing when the trigger is focused to avoid triggering a re-open animation
+          // on pointer interaction.
+          if (event.target !== subContext.trigger) context.onOpenChange(false);
+        })}
+        onEscapeKeyDown={composeEventHandlers(
+          props.onEscapeKeyDown,
+          (event) => {
+            rootContext.onClose();
+            // Ensure pressing escape in submenu doesn't escape full screen mode
+            event.preventDefault();
+          },
+        )}
+        onKeyDown={composeEventHandlers(props.onKeyDown, (event) => {
+          // Submenu key events bubble through portals. We only care about keys in this menu.
+          const isKeyDownInside = event.currentTarget.contains(
+            event.target as HTMLElement,
+          );
+          const isCloseKey = SUB_CLOSE_KEYS.includes(event.key);
+          if (isKeyDownInside && isCloseKey) {
+            context.onOpenChange(false);
+            // We focus manually because we prevented it in `onCloseAutoFocus`
+            subContext.trigger?.focus();
+            // Prevent window from scrolling
+            event.preventDefault();
+          }
+
+          /* When submenu opens to the left, we allow closing it with ArrowRight */
+          const side = context.content?.dataset.side as SubMenuSide;
+          if (
+            side === "left" &&
+            isKeyDownInside &&
+            event.key === "ArrowRight"
+          ) {
+            context.onOpenChange(false);
+            subContext.trigger?.focus();
+            event.preventDefault();
+          }
```

Comment:
Consider refactoring. Maybe combine the `if`s? Also inconsistent that the first code uses `SUB_CLOSE_KEYS` while the other has hard coded "ArrowRight".

Reply:
Done 🚀  Felt it added extra complexity to extract "event.key === "ArrowRight"" into a separate const, so kept it inline.

Reply:
Nice! Maybe you misunderstood the last sentence. Right now, arrow left is check with `SUB_CLOSE_KEYS.includes(event.key)` while arrow right is checked with `event.key === "ArrowRight"`. I think one could argue that it's a bit inconsistent. **Not a big deal though!**